enterprise/backend/src/metabase_enterprise/serialization/dump.clj

```diff
@@ -24,11 +25,41 @@
 
 (set! *warn-on-reflection* true)
 
+(def ^:private serialization-order
+  (delay (-> (edn/read-string (slurp (io/resource "serialization-order.edn")))
+             (update-keys name)
+             (update-vals (fn [order]
+                            (into {} (map vector order (range))))))))
+
+(def ^:private serialization-sorted-map
+  (memoize
+   (fn [t]
+     (if-let [order (get @serialization-order t)]
+       ;; known columns are sorted by their order, then unknown are sorted alphabetically
+       (let [getter #(if (contains? order %)
+                       [0 (get order %)]
+                       [1 %])]
+         (sorted-map-by (fn [k1 k2]
+                          (compare (getter k1) (getter k2)))))
+       (sorted-map)))))
+
+(defn- serialization-deep-sort [m]
+  (let [t    (-> (:serdes/meta m) last :model)
```

Comment:
This should be `some->`; this file is still technically used by v1 serdes as well as v2, and there's no `:serdes/meta` in v1. I think with `some->` it'll Just Work for v1: passing `nil` to `serialization-sorted-map` and getting back a plain `sorted-map` for alphabetic order.

Reply:
IDK, I prefer nil-punning to `some->` when it works:

```
user> (-> nil last :model)
nil
```